viewing_party/party.py

```diff
@@ -1,23 +1,157 @@
+    #comparing user movies to duplicates and creating a list of unique movies
+    user_unique_movies = []
+    for movies_watched in user_watched_data["watched"]:
+        if movies_watched not in user_duplicate_movies:
+            user_unique_movies.append(movies_watched)
+    return user_unique_movies
+
+def get_friends_unique_watched(user_watched_data):
+    #comparing friends movies to user movies and adding 
+    #unique entries to list
+    friend_unique_movies = []
+    for friends in user_watched_data["friends"]:
+        for friend, friends_movies in friends.items():
+            for movies in friends_movies:
+                if movies not in user_watched_data["watched"]:
+                    friend_unique_movies.append(movies)
+    friend_unique_movies_no_duplicates=[]
+    #removing the duplicate entry
+    for i in range(len(friend_unique_movies)): 
+        if friend_unique_movies[i] not in friend_unique_movies[i + 1:]: 
+            friend_unique_movies_no_duplicates.append(friend_unique_movies[i])
+    return friend_unique_movies_no_duplicates
+
+                    
 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_watched_data):
+    #retrieve unique movies that friends have watched
+    friend_unique_movies = get_friends_unique_watched(user_watched_data)
+    #create list of recs that only comes from subscriptions availible to user
+    recomendations = []
+    for friends in user_watched_data["friends"]:
+        for friend, friends_movies in friends.items():
+            for movies in friends_movies:
+                if movies in friend_unique_movies and movies["host"] in user_watched_data["subscriptions"]:
+                    recomendations.append(movies)
+    return recomendations
+
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_watched_data):
+    #retrieve unique movies that friends have watched
+    friend_unique_movies = get_friends_unique_watched(user_watched_data)
+    #create list of recs that only comes from subscriptions availible to user
+    recomendations = []
+    for key in user_watched_data:
+        # no subscriptions then we can't watch any recs
+        if key == "subscriptions":
+            for friends in user_watched_data["friends"]:
+                for friend, friends_movies in friends.items():
+                    for movies in friends_movies:
+                        if movies in friend_unique_movies and movies["host"] in user_watched_data["subscriptions"]:
+                            recomendations.append(movies)
+    # get genre fantasy recomendations
+    fantasy_recomendations = []
+    for rec in recomendations:
+        if rec["genre"] == "Fantasy":
+            fantasy_recomendations.append(rec)
+    return fantasy_recomendations

```

Comment:
How would you determine the user's most frequent genre if not always "Fantasy"? Hint- a previously written function would be a useful helper function.